swagger.yaml

```diff
@@ -132,141 +135,137 @@ paths:
         500:
           $ref: '#/responses/InternalError'
 
-  /images/{image_id}/upload:
-    post:
+  /images/{image_id}/downloads:
+    get:
       tags:
         - "image"
-      summary: "Upload an image"
-      description: "Supplies the uploaded file location and requests that the image be imported. This call sets the image state to 'uploaded'."
-      parameters:
-        - $ref: '#/parameters/image_id'
-        - $ref: '#/parameters/upload_image'
       produces:
         - "application/json"
       security:
         - FlorenceAPIKey: []
+        - ServiceAPIKey: []
+      parameters:
+        - $ref: '#/parameters/image_id'
       responses:
         200:
-          description: "The image upload was successfully requested to the image importer."
-        400:
-          description: "Invalid request, image id was incorrect"
+          description: "Successfully got download variant."
+          schema:
+            $ref: '#/definitions/ImageDownloads'
         401:
           $ref: '#/responses/Unauthenticated'
         403:
-          description: "Unauthorised to upload image or it is in wrong state to be uploaded"
+          description: "Unauthorised to view images metadata"
         404:
           $ref: '#/responses/NotFound'
         500:
           $ref: '#/responses/InternalError'
-
-  /images/{image_id}/publish:
     post:
       tags:
-      - "image"
-      summary: "Publish an image"
-      description: "Requests an image publishing via the static file publisher, which puts the S3 objects for this image to the static bucket. This call sets the image state to 'publishing'."
-      parameters:
-      - $ref: '#/parameters/image_id'
+        - "image"
       produces:
-      - "application/json"
+        - "application/json"
       security:
-      - FlorenceAPIKey: []
+        - FlorenceAPIKey: []
+        - ServiceAPIKey: []
+      parameters:
+        - $ref: '#/parameters/image_id'
+        - $ref: '#/parameters/new_image_download'
       responses:
-        200:
-          description: "The image publishing was successfully requested to Static file publisher."
+        201:
+          description: "Successfully created download variants."
+          schema:
+            $ref: '#/definitions/ImageDownload'
         400:
-          description: "Invalid request, image id was incorrect"
+          description: "Invalid request"
         401:
           $ref: '#/responses/Unauthenticated'
         403:
-          description: "Unauthorised to publish image or it is in wrong state to be published"
+          description: "Unauthorised to view images metadata"
         404:
           $ref: '#/responses/NotFound'
         500:
           $ref: '#/responses/InternalError'
 
   /images/{image_id}/downloads/{variant}:
-    put:
+    get:
```

Comment:
The code doesn't appear to have a `get` for `/images/{image_id}/downloads/{variant}`. Does it need adding in the code, or removing from the swagger spec?

Reply:
Ignore this comment - I have realised that this is in another PR 👍